### PR TITLE
allow empty service_account_private_key in GCE driver

### PR DESCRIPTION
**Pull request by jgleissner**
### What does this PR do?
This PR modifies the cloud GCE driver to allow an empty service_account_private_key parameter. Not passing a key to libcloud will make it use instance credentials for authentication. This is useful if you want to run salt-cloud on a GCE instance.

Instead of an empty string it would also be possible to use an explicit string like `use-instance-role-credentials` as the EC2 driver does for instance, if preferable.

### Tests written?

No

### Commits signed with GPG?

Yes

### CODE REVIEW

**Review by rallytime — APPROVED** [review]
Hi @jgleissner and welcome!

Can you add some documentation about this change?

**Review by rallytime — COMMENTED** [review]
Er, that last comment ^ needs to be resolved first. I accidentally hit approve. :D

**Comment by jgleissner** [comment]
I pushed a commit with some documentation about the feature. Is that the right place?

**Review by cachedout — CHANGES_REQUESTED** [review]
I would like to see something added to the release notes as well before we merge this, please.

**Comment by jgleissner** [comment]
I've added a note to the release notes file. Hope I did it ok. If not, let me know. Thanks!

**Review by gtmanfred — APPROVED** [review]

File: doc/topics/releases/fluorine.rst [hunk 1]
```diff
@@ -1315,3 +1315,13 @@ Proxy Modules
 - :mod:`salt.proxy.netmiko <salt.proxy.netmiko_px>`
 - :mod:`salt.proxy.nxos_api <salt.proxy.nxos_api>`
 - :mod:`salt.proxy.pyeapi <salt.proxy.arista_pyeapi>`
+
+Salt Cloud Features
```
Comment (rallytime) [inline doc/topics/releases/fluorine.rst]: Hey @jgleissner - Apologies for not being clear about this before, but can you move these to the `neon.rst` release notes? That's the codename for the current `develop` branch.

Comment (jgleissner) [reply doc/topics/releases/fluorine.rst]: I moved it to `neon.rst`. Seems I misunderstood the version_number readme. Sorry about that. Thanks for your patience. :)

**Review by rallytime — APPROVED** [review]
Thanks @jgleissner!

**Pull request merged**